### PR TITLE
Remove message of checking `apex.amp` module and add tests for features of gradient accumulation/mixed precision training

### PR DESCRIPTION
**Pull request by NaleRaphael**
This PR is a fix for issue #45 with some new test cases for those features implemented in PR #9.

A quick summary for this PR:
1. The message `To enable mixed precision training, please install apex...` is removed. (solved in commit 227fc53)
2. A silly mistake was made. `batch_size` was not passed into `DataLoader`, so that those data loaders in test cases were working with the default value `batch_size=1` before. Though it does not affect the test correctness, it still needs to be corrected. (solved in commit 1c549ec)
3. Another mistake. In test cases, there is no setting that moves model to device declared in `task.__init__()`. Therefore, all tests were running on CPU even if the pytest argument `--cpu_only` is not specified. (solved in commit c854714)
4. More tests for features of gradient accumulation and mixed precision are added in commit e073ac8.

Note that there is a new dependency `pytest-mock` added for new test cases.

### CODE REVIEW

File: tests/test_lr_finder.py [hunk 1]
```diff
@@ -4,6 +4,14 @@
 import task as mod_task
 
 
+try:
+    import apex
```
Comment (davidtvs) [inline tests/test_lr_finder.py:8]: I think this line can be changed to `from apex import amp` and we can then remove the local imports from the functions

File: tests/test_lr_finder.py [hunk 2]
```diff
@@ -90,6 +98,79 @@ def test_exponential_lr_history(self):
...
+        # So that the actual times of model/criterion called should be
+        # `(desired_bs/real_bs) * num_iter` == `accum_steps * num_iter`
+        assert spy.call_count == accum_steps * num_iter
+
+    @pytest.mark.skipif(
+        not (IS_APEX_AVAILABLE and mod_task.use_cuda()),
+        reason="`apex` module and gpu is required to run this test."
+    )
+    def test_gradient_accumulation_with_apex_amp(self, mocker):
+        from apex import amp
```
Comment (davidtvs) [inline tests/test_lr_finder.py:124]: Remove line (see comment about import)

File: tests/test_lr_finder.py [hunk 3]
```diff
@@ -90,6 +98,79 @@ def test_exponential_lr_history(self):
...
+        assert spy.call_count == accum_steps * num_iter
+
+
+@pytest.mark.skipif(
+    not (IS_APEX_AVAILABLE and mod_task.use_cuda()),
+    reason="`apex` module and gpu is required to run these tests."
+)
+class TestMixedPrecision:
+    def test_mixed_precision(self, mocker):
+        from apex import amp
```
Comment (davidtvs) [inline tests/test_lr_finder.py:152]: Remove line (see comment about import)

File: tests/test_lr_finder.py (hunk 1 above)
Comment (NaleRaphael) [reply tests/test_lr_finder.py:8]: Yeah, I'll fix it.
